hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaCreationToOutputScriptTest.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.schemaupdate;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.boot.spi.MetadataImplementor;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Environment;
+import org.hibernate.dialect.H2Dialect;
+import org.hibernate.engine.config.spi.ConfigurationService;
+import org.hibernate.service.ServiceRegistry;
+import org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator;
+
+import org.hibernate.testing.RequiresDialect;
+import org.hibernate.testing.ServiceRegistryBuilder;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseUnitTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.StringContains.containsString;
+
+@TestForIssue(jiraKey = "HHH-11817")
+@RequiresDialect(H2Dialect.class)
+public class SchemaCreationToOutputScriptTest extends BaseUnitTestCase {
+
+	private final String createTableMyEntity = "create table MyEntity";
+	private final String createTableMySecondEntity = "create table MySecondEntity";
+
+	private File output;
+	private ServiceRegistry serviceRegistry;
+	private MetadataImplementor metadata;
+
+	@Before
+	public void setUp() throws Exception {
+		output = File.createTempFile( "creation_script", ".sql" );
+		output.deleteOnExit();
+
+		List<String> content = Arrays.asList(
+				"This is the database creation script generated by Hibernate"
+				, "This is the second line"
+				, "This is the third line"
+				, "This is the fourth line"
+				, "This is the fifth line"
+				, "This is the sixth line"
+				, "This is the seventh line"
+				, "This is the eighth line"
+				, "This is the ninth line"
+		);
+
+		try (BufferedWriter bw = new BufferedWriter( new FileWriter( output ) )) {
+			for ( String s : content ) {
+				bw.write( s );
+				bw.write( System.lineSeparator() );
```

Comment:
You might well use another decorator layer if what is written is string
```
try (PrintWriter pw = new PrintWriter(new BufferedWriter( new FileWriter( output ) ) ) ) {
    for ( String s : content ) {
        pw.println( s );
    }
}
```
but let us stop here. What really matters is the testing per se!
Whenever I am in an interview and asked about design pattern, I wanna say "there is good example of design pattern overuse in Java - Legacy IO based on decorator pattern!".

Reply:
Personally not at all concerned with what tests do so long as they verify the expected behavior